pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,337 @@
+	}
+
+	Int disk_space = ceil(2.2 * size(bam, "GB")) + 50
+
+	command <<<
+		fgbio ExtractUmisFromBam --input ~{bam} \
+			--read-structure ~{read1Structure} \
+			--read-structure ~{read2Structure} \
+			--molecular-index-tags RX \
+			--output extractUMIs.out.bam
+	>>>
+
+	runtime {
+		docker : "quay.io/biocontainers/fgbio@sha256:a8e5cf58c318bffba3b2b694a3640ecd9e8106cee2e33b75710c0e8215138b6e"
+		disks : "local-disk " + disk_space + " HDD"
+		preemptible: 0
+	}
+
+	output {
+		File bam_umis_extracted = "extractUMIs.out.bam"
+	}
+}
+
+task rnaseqc2 {
+	input {
+		File bam_file
+		File genes_gtf
+		String sample_id
+	}
+	Int disk_space = ceil(size(bam_file, 'GB') + size(genes_gtf, 'GB')) + 100
+
+	command {
+		set -euo pipefail
+		echo $(date +"[%b %d %H:%M:%S] Running RNA-SeQC 2")
+		rnaseqc ~{genes_gtf} ~{bam_file} . -s ~{sample_id} -vv
+		echo "  * compressing outputs"
+		gzip *.gct
+		echo $(date +"[%b %d %H:%M:%S] done")
+	}
+
+	output {
+		File gene_tpm = "${sample_id}.gene_tpm.gct.gz"
+		File gene_counts = "${sample_id}.gene_reads.gct.gz"
+		File exon_counts = "${sample_id}.exon_reads.gct.gz"
+		File metrics = "${sample_id}.metrics.tsv"
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/ckachulis/rnaseqc:2.4.2"
+		memory: "10GB"
+		disks: "local-disk " + disk_space + " HDD"
+		preemptible: 0
+	}
+}
+
+task GetSampleName {
+	input {
+		File bam
+	}
+
+	parameter_meta {
+		bam : {
+			localization_optional : true
+		}
+	}
+
+	command <<<
+		gatk GetSampleName -I ~{bam} -O sample_name.txt
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-gatk/gatk:4.2.0.0"
+		disks: "local-disk 100 HDD"
+	}
+
+	output {
+		String sample_name = read_string("sample_name.txt")
+	}
+}
+
+task CopyReadGroupsToHeader {
+	input {
+		File bam_with_readgroups
+		File bam_without_readgroups
+	}
+
+	Int disk_size = ceil(2.0 * size([bam_with_readgroups, bam_without_readgroups], "GB")) + 10
+	String basename = basename(bam_without_readgroups)
+
+	command <<<
+		samtools view -H ~{bam_without_readgroups} > header.sam
+		samtools view -H ~{bam_with_readgroups} | grep "@RG" >> header.sam
+		samtools reheader header.sam ~{bam_without_readgroups} > ~{basename}
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/samtools@sha256:0e49b0a5d91c203b8c07f5242277c2060b4b8ea54df8b1d123f990a1ad0588b2"
+		disks: "local-disk " + disk_size + " HDD"
+	}
+
+	output {
+		File output_bam = basename
+	}
+}
+
+task CollectRNASeqMetrics {
+	input {
+		File input_bam
+		File input_bam_index
+		String output_bam_prefix
+		File ref_dict
+		File ref_fasta
+		File ref_fasta_index
+		Int preemptible_tries
+		File ref_flat
+		File ribosomal_intervals
+	}
+
+	Float ref_size = size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB") + size(ref_dict, "GiB")
+	Int disk_size = ceil(size(input_bam, "GiB") + ref_size) + 20
+
+	# This jar skips the header check of the ribosomal interval
+	File picard_jar = "gs://broad-dsde-methods-takuto/hydro.gen/picard_ignore_ribosomal_header.jar"
```

Comment:
It turns out this specific docker was being used because Takuto had some testing input files that had non-standard headers. We have gone back to using a 'standard' picard docker and the pipeline works.